### PR TITLE
feat: Add surveys to remote config

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Would be cool if we could stop hammering the API for no good reason.

## Changes

* Starts adding surveys to remote config
* Probs makes sense to add corresponding JS code first to check it all works as expected (incoming)

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: posthog/api/survey.py [hunk 1]
```diff
@@ -801,6 +801,19 @@ class Meta:
         fields = ["survey_config"]
 
 
+class SurveyAPIActionSerializer(serializers.ModelSerializer):
```
Comment (marandaneto) [inline posthog/api/survey.py:804]: that was added in https://github.com/PostHog/posthog/pull/26759/ already
probably just need to merge master into this branch?

File: posthog/models/remote_config.py [hunk 2]
```diff
@@ -177,8 +179,12 @@ def build_config(self):
                 config["quotaLimited"] = ["recordings"]
                 config["sessionRecording"] = False
 
-        config["surveys"] = True if team.surveys_opt_in else False
         config["heatmaps"] = True if team.heatmaps_opt_in else False
+        surveys_response = get_surveys_response(team)
+        config["surveys"] = surveys_response["surveys"]
+        if surveys_response["survey_config"]:
+            config["survey_config"] = surveys_response["survey_config"]
```
Comment (marandaneto) [inline posthog/models/remote_config.py:186]:
```suggestion
        survey_config = surveys_response["survey_config"]
        if survey_config:
            config["survey_config"] = survey_config
```

nitpick but I tend to assign to a var so we don't look up the map twice, it's more important when the map has many keys so not a must here, your call.

File: posthog/models/remote_config.py [hunk 3]
```diff
@@ -177,8 +179,12 @@ def build_config(self):
                 config["quotaLimited"] = ["recordings"]
                 config["sessionRecording"] = False
 
-        config["surveys"] = True if team.surveys_opt_in else False
         config["heatmaps"] = True if team.heatmaps_opt_in else False
+        surveys_response = get_surveys_response(team)
+        config["surveys"] = surveys_response["surveys"]
```
Comment (marandaneto) [inline posthog/models/remote_config.py:184]: The JS SDK expects a boolean. Will that work with the new backend and the old JS SDK version? (not everyone uses the loader script).

**Review by marandaneto — APPROVED** [review]
Just a question about a possible breaking change otherwise LGTM

File: posthog/models/remote_config.py (hunk 2 above)
Comment (benjackwhite) [reply posthog/models/remote_config.py:186]: I'd debate this has almost zero perf impact 🤷 will ignore in favour of getting this in

Comment (marandaneto) [reply posthog/models/remote_config.py:186]: agree!

> it's more important when the map has many keys so not a must here

:)

File: posthog/models/remote_config.py (hunk 3 above)
Comment (marandaneto) [reply posthog/models/remote_config.py:184]: https://github.com/PostHog/posthog-js/blob/99e92b5f01858f4fc4584e8c6b45269ea432d8e1/src/posthog-surveys.ts#L73
at least the key is being read here!

**Pull request merged**